src/Build/BuildCheck/Infrastructure/TracingReporter.cs

```diff
@@ -24,4 +31,15 @@ public void AddStats(string name, TimeSpan subtotal)
             TracingStats[name] = subtotal;
         }
     }
+
+    public void AddAnalyzerInfraStats()
+    {
+        var infraStats = new Dictionary<string, TimeSpan>() {
```

Comment:
This is non-blocking but storing a known list of things in a dictionary feels a bit weird to me.

Reply:
Yeah, just using a dictionary since we already had the methods for it. If it a bother I can write some new methods to go with another structure.